habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -191,15 +191,18 @@ def _get_observation_space(self, *args: Any, **kwargs: Any):
         return spaces.Box(
             low=np.iinfo(np.uint32).min,
             high=np.iinfo(np.uint32).max,
-            shape=(self.config.HEIGHT, self.config.WIDTH),
-            dtype=np.uint32,
+            shape=(self.config.HEIGHT, self.config.WIDTH, 1),
+            dtype=np.int32,
         )
 
     def get_observation(
         self, sim_obs: Dict[str, Union[np.ndarray, bool, "Tensor"]]
     ) -> VisualObservation:
         obs = cast(Optional[VisualObservation], sim_obs.get(self.uuid, None))
         check_sim_obs(obs, self)
+        # make semantic observation a 3D array
+        if isinstance(obs, np.ndarray):
+            obs = obs[..., None].astype(np.int32)
```

Comment:
```suggestion
            obs = obs[..., None].astype(np.int32)
    else:
            obs = obs[..., None]
```
PyTorch tensors can still be returned as observations if the GPU2GPU flag is enabled

Reply:
@Skylion007 the mypy tests fail if I handle torch tensor. It says the statement is unreachable that's why I removed change on L207

Reply:
@Ram81 An easy way to get this merged is to do obs = obs[..., None] before the isinstance and just do the as type after